sgt-back-end/server.js

```diff
@@ -0,0 +1,149 @@
+import pg from 'pg';
+import express from 'express';
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+const app = express();
+app.use(express.json());
+
+// GET STARTS HERE
+
+app.get('/api/grades', async (req, res) => {
+  try {
+    const sql = `
+    select *
+      from "grades"
+      `;
+    const result = await db.query(sql);
+    const allGrades = result.rows;
+    if (allGrades) {
+      res.status(200).json(allGrades);
+    } else {
+      res.status(404).json({ error: 'Cannot find grades' });
+    }
+  } catch (err) {
+    console.error(err);
+    res.status(500).json({ error: 'An unexpected error occured' });
+  }
+});
+
+// GET ENDS HERE
+
+// POST STARTS HERE
+
+app.post('/api/grades', async (req, res) => {
+  try {
+    const newEntryName = req.body.name;
+    const newEntryCourse = req.body.course;
+    const newEntryScore = Number(req.body.score);
+    if (newEntryCourse === undefined || newEntryName === undefined || newEntryScore === undefined) {
```

Comment:
Because of the `Number` conversion above, `newEntryScore` will always be defined, but will be NaN if `req.body.score` doesn't exist. Safest is to check `!newEntryScore.isInteger()` here instead of checking against `undefined`.